doc/user_guide/recordflux_ug/03-getting_started_with_recordflux.rst

```diff
@@ -0,0 +1,743 @@
+
+The automated correctness proof performed at the specification level
+(i.e. before generating any source code) guarantee the following
+properties:
+
+   * Determinism (no contradictory conditions)
+   * Liveness (no cycles)
+   * Reachability (no unused fields)
+   * Coherency (no overlaps)
+   * Completeness (no holes)
+
+Consequently, the resulting message parsers and message serializers
+are -- by construction -- free from many issues that plague manual
+implementations of communication protocols. As functional correctness
+is proven statically, parsers guarantee that received messages comply
+with their specification, serializers ensure the creation of correct
+messages, and state machines provide the specified protocol behavior.
+
+The extensive guarantees provided by the generated code do not degrade
+performance: the results are usable in resource-constrained,
+deeply-embedded systems.
+
+
+.. rubric:: AdaCore tool stack
+
+RecordFlux is part of AdaCore's tool stack for creating high-assurance
+implementations of binary data formats and communication protocols.
+At the highest level, the RecordFlux DSL is used to formalize the
+format and dynamic behavior. The DSL is usable by domain experts who
+need not be programmers, and the RecordFlux tool set can be used to
+statically prove important properties of the formal specification.
+
+While a formal high-level model is worthwhile on its own, connecting
+it to an actual implementation allows extending the properties of the
+model into the source code. In the case of AdaCore's SPARK code
+generation, the process is even more rigorous: high-level model
+properties are translated into provable SPARK code in such a way that
+the absence of run-time errors can be automatically proven.
+In addition, it is possible to demonstrate key functional properties
+derived from the specification.
+
+As a last step in the tool stack, source code is compiled into a binary
+executable for the target CPU. Code generated by RecordFlux can be
+compiled by GNAT Pro Assurance, AdaCore's complete solution for projects
+with the most stringent requirements for reliability,
+long-term maintenance, or certification. Only a minimum of run-time
+features are necessary, which allows deploying the code on platforms
+ranging from full-blown servers down to embedded bare-metal targets.
+The compiler hardening options provided by GNAT Pro Assurance can be
+used to further mitigate attacks on network-facing protocol handling code.
+
+.. _TLV_Example:
+
+Example
+=======
+
+.. index:: TLV Example
+
+The example below shows how to use RecordFlux to define the layout of a
+binary message, verify its syntax, generate code to construct and
+parse messages, formally prove that the generated code is free of
+run-time errors, and conduct a session that processes a sample
+message.
+
+The source code for the example is in the file :file:`tlv.rflx` in the
+:file:`examples/tlv/` directory of the RecordFlux distribution.
```

Comment:
It is not distributed when installing via `pip`. Maybe we should reference the source repository?